backend/tournesol/tests/test_exports.py

```diff
@@ -337,100 +369,86 @@ def test_all_exports_voting_rights(self):
+
+        for expected_export, row in zip(expected_exports, rows):
+            row = rows.pop(0)
+            self.assertEqual(
+                row["public_username"],
+                expected_export.contributor_rating.user.username,
+            )
+            self.assertEqual(
+                row["video"],
+                expected_export.contributor_rating.entity.metadata["video_id"],
+            )
+            self.assertEqual(row["criteria"], expected_export.criteria)
+            self.assertEqual(row["score"], str(expected_export.score))
+            self.assertEqual(row["voting_right"], str(expected_export.voting_right))
+
+    @export_test_override_settings
     def test_all_export_sorts_by_username(self):
         last_user = UserFactory(username="z")
         first_user = UserFactory(username="a")
-
         self.add_comparison(user=last_user, is_public=True)
         self.add_comparison(user=first_user, is_public=True)
 
         call_command("create_dataset")
-
         response = self.client.get("/exports/all/")
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-
-        content_disposition = response.headers["Content-Disposition"]
-        match = re.search("filename=(.+).zip", content_disposition)
-        root = match.group(1)
-
-        zip_content = io.BytesIO(response.content)
-
-        with zipfile.ZipFile(zip_content, "r") as zip_file:
-            with zip_file.open(root + "/users.csv", "r") as file:
-                content = file.read().decode("utf-8")
-                csv_file = csv.DictReader(io.StringIO(content))
-                rows = list(csv_file)
-
-                usernames = [row["public_username"] for row in rows]
-                self.assertEqual("a", usernames[0])
-                self.assertEqual("z", usernames[-1])
+        rows = self.parse_zipped_csv(response, "users.csv")
+        usernames = [row["public_username"] for row in rows]
+        self.assertEqual("a", usernames[0])
+        self.assertEqual("z", usernames[-1])
+
+    @export_test_override_settings
+    def test_all_export_contains_collective_criteria_scores(self):
+        EntityCriteriaScoreFactory(criteria="h2g2", score=42, entity__metadata__name="DON'T PANIC")
+        call_command("create_dataset")
+        response = self.client.get("/exports/all/")
+        rows = self.parse_zipped_csv(response, "collective_criteria_scores.csv")
+        self.assertEqual(len(rows), 1)
+        self.assertEqual(float(rows[0]["score"]), 42.)
+        self.assertEqual(rows[0]["criteria"], "h2g2")
+        self.assertEqual(rows[0]["name"], "DON'T PANIC")
+
+    @export_test_override_settings
+    def test_collective_criteria_scores_is_empty_without_criteria_scores(self):
+        EntityCriteriaScore.objects.all().delete
+        call_command("create_dataset")
+        response = self.client.get("/exports/all/")
+        rows = self.parse_zipped_csv(response, "collective_criteria_scores.csv")
+        self.assertEqual(len(rows), 0)
```

Comment:
These are the new tests for the collective_criteria_scores. Do you want to add more?

Reply:
Perfect, these tests look good to me : )

We can check the presence of each expected metadata column with `self.assertEqual`, but first we should investigate what kind of metadata we are allowed to export.